pdata/plog/plogotlp/logs.go

```diff
@@ -0,0 +1,164 @@
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+var jsonMarshaler = &jsonpb.Marshaler{}
+var jsonUnmarshaler = &jsonpb.Unmarshaler{}
+
+// LogsResponse represents the response for gRPC client/server.
+type LogsResponse struct {
+	orig *otlpcollectorlog.ExportLogsServiceResponse
+}
+
+// NewLogsResponse returns an empty LogsResponse.
+func NewLogsResponse() LogsResponse {
+	return LogsResponse{orig: &otlpcollectorlog.ExportLogsServiceResponse{}}
+}
+
+// MarshalProto marshals LogsResponse into proto bytes.
+func (lr LogsResponse) MarshalProto() ([]byte, error) {
+	return lr.orig.Marshal()
+}
+
+// UnmarshalProto unmarshalls LogsResponse from proto bytes.
+func (lr LogsResponse) UnmarshalProto(data []byte) error {
+	return lr.orig.Unmarshal(data)
+}
+
+// MarshalJSON marshals LogsResponse into JSON bytes.
+func (lr LogsResponse) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	if err := jsonMarshaler.Marshal(&buf, lr.orig); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// UnmarshalJSON unmarshalls LogsResponse from JSON bytes.
+func (lr LogsResponse) UnmarshalJSON(data []byte) error {
+	return jsonUnmarshaler.Unmarshal(bytes.NewReader(data), lr.orig)
+}
+
+// LogsRequest represents the response for gRPC client/server.
+type LogsRequest struct {
+	orig *otlpcollectorlog.ExportLogsServiceRequest
+}
+
+// NewLogsRequest returns an empty LogsRequest.
+func NewLogsRequest() LogsRequest {
+	return LogsRequest{orig: &otlpcollectorlog.ExportLogsServiceRequest{}}
+}
+
+// MarshalProto marshals LogsRequest into proto bytes.
+func (lr LogsRequest) MarshalProto() ([]byte, error) {
+	return lr.orig.Marshal()
+}
+
+// UnmarshalProto unmarshalls LogsRequest from proto bytes.
+func (lr LogsRequest) UnmarshalProto(data []byte) error {
+	if err := lr.orig.Unmarshal(data); err != nil {
+		return err
+	}
+	otlp.InstrumentationLibraryLogsToScope(lr.orig.ResourceLogs)
+	return nil
+}
+
+// MarshalJSON marshals LogsRequest into JSON bytes.
+func (lr LogsRequest) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	if err := jsonMarshaler.Marshal(&buf, lr.orig); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// UnmarshalJSON unmarshalls LogsRequest from JSON bytes.
+func (lr LogsRequest) UnmarshalJSON(data []byte) error {
+	if err := jsonUnmarshaler.Unmarshal(bytes.NewReader(data), lr.orig); err != nil {
+		return err
+	}
+	otlp.InstrumentationLibraryLogsToScope(lr.orig.ResourceLogs)
+	return nil
+}
+
+func (lr LogsRequest) SetLogs(ld plog.Logs) {
+	lr.orig.ResourceLogs = internal.LogsToOtlp(ld).ResourceLogs
+}
+
+func (lr LogsRequest) Logs() plog.Logs {
+	return internal.LogsFromOtlp(&otlplogs.LogsData{ResourceLogs: lr.orig.ResourceLogs})
+}
+
+// LogsClient is the client API for OTLP-GRPC Logs service.
+//
+// For semantics around ctx use and closing/ending streaming RPCs, please refer to https://godoc.org/google.golang.org/grpc#ClientConn.NewStream.
+type LogsClient interface {
```

Comment:
should this just be `plogotlp.Client` (and `plogotlp.Server`), similar to other changes suggested?
Same question for other signals.

Reply:
This sounds reasonable to me. I think we can rename `plogotlp.Client` and `plogotlp.Server` but we should keep `plogotlp.LogsResponse` and `plogotlp.LogsRequest` in this package as is. Let me know if this is what you meant.

Reply:
I would also lean towards `plogotlp.Request` and `plogotlp.Response`, but I don't feel quite as strongly about that.

Reply:
I like dropping the "Logs" prefix for Request/Response.

Reply:
I was thinking that removing `Logs` prefix from `Request` and `Response` would leave it without enough connection to `plog.Logs`, but probably `Request.SetLogs` and `Request.Logs` is enough. So I'm ok with removing prefix as well.

Reply:
They will always be "referenced" as "plogotlp.Request" when using the type (based on golang rule).